pkg/storaged/content-views.jsx

```diff
@@ -842,17 +872,68 @@ export class VGroup extends React.Component {
+                                   min: 5 * 1024 * 1024 * 1024,
+                                   max: vgroup.FreeSize,
+                                   round: vgroup.ExtentSize
+                               }),
+                    SizeSlider("vdo_lsize", _("Logical Size"),
+                               {
+                                   visible: vals => vals.purpose === 'vdo',
+                                   value: vgroup.FreeSize,
+                                   // visually point out that this can be over-provisioned
+                                   max: vgroup.FreeSize * 3,
+                                   allow_infinite: true,
+                                   round: vgroup.ExtentSize
+                               }),
+
+                    CheckBoxes("vdo_options", _("Options"),
+                               {
+                                   visible: vals => vals.purpose === 'vdo',
+                                   fields: [
+                                       {
+                                           tag: "compression", title: _("Compression"),
+                                           tooltip: _("Save space by compressing individual blocks with LZ4")
+                                       },
+                                       {
+                                           tag: "deduplication", title: _("Deduplication"),
+                                           tooltip: _("Save space by storing identical data blocks just once")
+                                       },
+                                   ],
+                                   value: {
+                                       compression: true,
+                                       deduplication: true,
+                                   }
+                               }),
                 ],
                 Action: {
                     Title: _("Create"),
-                    action: function (vals) {
+                    action: (vals, progress) => {
                         if (vals.purpose == "block")
                             return vgroup.CreatePlainVolume(vals.name, vals.size, { });
                         else if (vals.purpose == "pool")
                             return vgroup.CreateThinPoolVolume(vals.name, vals.size, { });
+                        else if (vals.purpose == "vdo") {
+                            return (need_vdo_install ? install_package(vdo_package, progress) : Promise.resolve())
+                                    .then(() => {
+                                        progress(_("Creating VDO device")); // not cancellable any more
+                                        return vgroup.CreateVDOVolume(
+                                            // HACK: hardcode pool name; https://github.com/storaged-project/udisks/issues/939
+                                            vals.name, vals.name + "_vdopool",
```

Comment:
We could re-implement the loguic in lvcreate here and choose the first 'vpool<N>' name that is still free, similar to how we create default names for new logical volumes.

Reply:
Good idea, I'll look into that.

Reply:
Conveniently there already is next_default_logical_volume_name(), so I just had to generalize this a bit. Done!